### DISCUSSION TITLE
[WASM] Resources not working on non-english speaking devices

### DISCUSSION THREAD

**Discussion by DierkDroth** [unoplatform/uno#8351] [Q&A]
~~My UNO WASM app (latest Uno 4.2 bits) is working properly (within the limitations/issues which I´m aware of) on~~
~~- Chrome on Windows 11~~
~~- Safari on Mac (MacBook Air)~~
~~However, as I try to start the app on my iPad (Safari on iOS) the rendering is completely screwed up (e.g. TextBlocks not rendering at all) and even some under-the-hood logic appears not to be working (no idea how to debug though).~~
~~UNO Gallery appears to be working on my iPad.~~
~~Are there known limitations with UNO WASM on Safari on iOS? Or do I have to apply some tricks to make sure UNO WASM on Safari on iOS works as it does with Safari on Mac?~~

The whole thread was going wrong direction: the problem is that string resources which are in the '<shared project>/Strings/**en-US**' directory, which works fine on english speaking devices, but not on non-english speaking devices. For whatever reason the resource logic would not fall back to the single available resource - which is 'en-US' in my case.

I fixed the problem by renaming the resource directory from ' en-US' to 'en'. Go figure...

**Comment by DierkDroth** [comment]
@jeromelaban the problem is actually more wide spread than I thought:
- the app (AOT build, latest UNO 4.2 bits) renders fine on my Chrome on Windows
- the same app does not render any TextBlocks on Chrome on a customer´s Windows machine

There appears to be a problem with loading the Roboto fonts (just guessing). I attached the Font.css file.

Would you have any idea what could be the source of trouble? 
Thanks in advance

[Fonts.zip](https://github.com/unoplatform/uno/files/8306205/Fonts.zip)

**Reply by jeromelaban** [reply]
This is not a known issue. If you have an app that reproduces the issue, it'll help troubleshoot.

**Reply by DierkDroth** [reply]
@jeromelaban I tried to reproduce - even with the custom theme, based on Material + Roboto - but failed.

Also, there is nothing in browser console.

I could PM you the link to the app (it´s far from being publically available) in question if that helps. Please let me know if you wanted to take a look.

**Reply by jeromelaban** [reply]
We may not be able to fix the issue if we can't reproduce on our end, unfortunately.

**Reply by DierkDroth** [reply]
Understand. I´m not even sure there is something for UNO to fix. This is rather trying to understand what **might** be the cause of trouble. 

Note: you even might not be able to reproduce it at all with any given repo, since it´s works for me on 2 completely different devices, but does not work on others. Meaning: there very likely is no scenario at all, where it could be reproduced close to 100%.

I´m open for any other idea you had.

**Reply by jeromelaban** [reply]
It could be related to https://github.com/unoplatform/uno/issues/693, then.

**Reply by DierkDroth** [reply]
Yes, absolutely. I saw this link before. However, it makes me wonder why it wouldn´t fall back to ¨Times New Roman" (that´s the browser´s default font, no?)? 

There is just not any text on TextBlocks and Buttons rendered. Interestingly: text is rendered for TextBox. May be since the FontSpacing is different?!? Go figure...

After all this look pretty much like a race condition: works great on my lighting fast DEV machine, but does not work on a slow customer´s machine...

**Reply by DierkDroth** [reply]
**Test #1**

@jeromelaban I setup a custom index.html and added the lines below  as per [here ](https://fonts.google.com/specimen/Roboto)in a attempt to 'force' loading the Roboto font. 
```
	<link rel="preconnect" href="https://fonts.googleapis.com">
	<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
	<link href="https://fonts.googleapis.com/css2?family=Roboto&display=swap" rel="stylesheet">
```
However, this did not help

**Reply by DierkDroth** [reply]
**Test #2**

Next I tried adding these lines instead (as alternate approach as per link above)

```
	<style>
		@import url('https://fonts.googleapis.com/css2?family=Roboto&display=swap');
	</style>
```

This did not change a thing neither

**Reply by DierkDroth** [reply]
**Test #3**

(updated)
@jeromelaban lastly I tested removing any 'Roboto' reference from my custom theming and also from Fonts.css and uploaded a FullAOT build to a different URL (to make sure caching issues aren't getting in my way).

I found:
- app renders properly on my DEV laptop (Chrome). The rendering with browser default font looks slightly - but noticible - different than when using Roboto font
- on my iPad **I still see the same problem: TextBlocks are not rendered**

This means the issue is not related to custom fonts at all, no? So why arent' even default fonts rendered?

**Reply by jeromelaban** [reply]
I do not know, and without a repro we cannot diagnose. Note that `FullAOT` is not supported [as of latest .NET builds](https://github.com/unoplatform/Uno.Wasm.Bootstrap#full-aot-mode).

**Reply by DierkDroth** [reply]
@jeromelaban thanks for follow up
- sorry for confusion about FullAOT. I checked: you're right, I see this in my project file `InterpreterAndAOT`
- I tried to explain that a repo will not help for various reasons:
-- even if would be able to have repo which would show the issue on 'some' devices, it will work OK on others - including yours (and mine)
-- I do not know what to cut away from my full blown app to create something 'reproducible' (whatever that means in the context what presumably is a race condition) + every iteration is painstakingly slow since a build + upload takes >= 30 minutes

There has to be another, viable way to tackle that show stopper

**Reply by DierkDroth** [reply]
**Test #4**

As next test I removed any custom theme from my app. This same problem.

So:
- it's not related to Material/custom theming
- it's not related to custom fonts (Roboto)

**Reply by DierkDroth** [reply]
**Test #5**

Next I downloaded the Uno.Gallery repo updated it to latest 4.2 bits + .NET 6 and uploaded it to my AWS S3 web space.

**Surprise, surprise, that worked as expected on any device I tested!**

This made me trying to understand how the fonts management in Uno.Gallery works and found that Fonts.css looks completely differrent than [documented here](https://platform.uno/docs/articles/features/custom-fonts.html) - and than in my app - and there is even is some custom logic around FontLoader.js (+ may be even more stuff to setup).

-> @jeromelaban I now really needed a working sample + documentation on how to properly manage (custom) fonts with UNO WASM so my app would work reliably as Uno.Gallery does.

Thanks in advance
